src/contract.rs

```diff
@@ -101,8 +107,75 @@ pub fn execute(
             timestamp_ct as usize,
         ),
 
-        ExecuteMsg::RemoveTimestamp { timestamp } => remove_timestamp(deps, info, timestamp)
+        ExecuteMsg::RemoveTimestamp { timestamp } => remove_timestamp(deps, info, timestamp),
+        ExecuteMsg::AddOffChainMetrics {off_chain_metrics} =>
+            {add_off_chain_metrics (deps, info, off_chain_metrics)}
+        ExecuteMsg::RemoveOffChainMetricsForTimestamp {timestamp} =>
+            {remove_off_chain_metrics_for_timestamp(deps, info, timestamp)}
+    }
+}
+
+//todo: test
+fn remove_off_chain_metrics_for_timestamp(deps: DepsMut, info: MessageInfo, timestamp: u64)
+    -> Result<Response, ContractError> {
+
+    if !sender_is_manager(&deps, &info) {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    let mut off_chain_timestamps = OFF_CHAIN_TIMESTAMPS.load(deps.storage)?;
+
+    let existing_timestamps_length = off_chain_timestamps.timestamps.len();
+
+    let new_timestamps = off_chain_timestamps.timestamps.into_iter()
+        .filter(|state_timestamp| { state_timestamp.ne(&timestamp)})
+        .collect::<Vec<u64>>();
+
+    let mut timestamp_existed = false;
+
+    if new_timestamps.len() < existing_timestamps_length {
+        off_chain_timestamps.timestamps = new_timestamps;
+        OFF_CHAIN_TIMESTAMPS.save(deps.storage, &off_chain_timestamps)?;
+        OFF_CHAIN_STATE.remove(deps.storage, U64Key::from(timestamp));
+        timestamp_existed = true;
     }
+
+    Ok(Response::new()
+        .add_attribute("method", "remove_off_chain_metrics_for_timestamp")
+        .add_attribute("message",
+                       if timestamp_existed {"timestamp successfully removed"}
+                       else {"timestamp didn't exist"}))
+}
+
+//todo:test
+fn add_off_chain_metrics(deps: DepsMut, info: MessageInfo, metrics_to_be_added: OffChainMetrics)
```

Comment:
1. We will have upto a 130 validators in 2-3 weeks from now. I think we will hit the message too long error if we try take on all of this data in one message.